sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,375 @@
+        TracerProvider tracerProvider = Mockito.spy(new TracerProvider(getMockTracer(mockTracer)));
+        ReflectionUtils.setTracerProvider(client, tracerProvider);
+        TracerProviderCapture tracerProviderCapture = new TracerProviderCapture();
+        Mockito.doAnswer(tracerProviderCapture).when(tracerProvider).startSpan(Matchers.anyString(),
+            Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+
+        cosmosAsyncContainer.getScripts().readAllStoredProcedures(new CosmosQueryRequestOptions()).byPage().single().block();
+        Context context = tracerProviderCapture.getResult();
+
+        Mockito.verify(tracerProvider, Mockito.times(1)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readAllStoredProcedures." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 1);
+
+        cosmosAsyncContainer.getScripts().readAllTriggers(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(2)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readAllTriggers." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 2);
+
+        cosmosAsyncContainer.getScripts().readAllUserDefinedFunctions(new CosmosQueryRequestOptions()).byPage().single().block();
+        Mockito.verify(tracerProvider, Mockito.times(3)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readAllUserDefinedFunctions." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 3);
+
+        CosmosUserDefinedFunctionProperties cosmosUserDefinedFunctionProperties =
+            getCosmosUserDefinedFunctionProperties();
+        CosmosUserDefinedFunctionProperties resultUdf =
+            cosmosAsyncContainer.getScripts().createUserDefinedFunction(cosmosUserDefinedFunctionProperties).block().getProperties();
+        Mockito.verify(tracerProvider, Mockito.times(4)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "createUserDefinedFunction." + cosmosAsyncContainer.getId(), context,
+            cosmosAsyncDatabase.getId(), 4);
+
+        cosmosAsyncContainer.getScripts().getUserDefinedFunction(cosmosUserDefinedFunctionProperties.getId()).read().block();
+        Mockito.verify(tracerProvider, Mockito.times(5)).startSpan(Matchers.anyString(), Matchers.anyString(),
+            Matchers.anyString(), Matchers.any(Context.class));
+        verifyTracerAttributes(mockTracer, "readUDF." + cosmosAsyncContainer.getId(), context,
```

Comment:
I thought we got rid of the span name `readUDF` and should be using the same public API name for user -API visibility concerns.

Reply:
we can do readUDF -> readUserDefinedFunction, but cant do just read() as we have hierarchy model in cosmos and all resources use read() api